straxen/plugins/nveto_pulse_processing.py

```diff
@@ -92,4 +93,27 @@ def clean_up_empty_records(records, only_last=True):
         if np.any(r['data'] != 0):
             indicies_to_keep[n_indicies] = ind
             n_indicies += 1
+            continue
+
+        # If we arrive here this means we will remove this record
+        # Hence we have to update the pulse_lengths accordingly:
+        length = r['length']
+
+        TRIAL_COUNTER = 500  # Do not want to be stuck forever
+        for neighbors in record_links:
+            neighbor_i = ind
+            ntries = 0
+
+            while ntries < TRIAL_COUNTER:
+                neighbor_i = neighbors[neighbor_i]
+                if neighbor_i == -1:
```

Comment:
This should be ```strax.NO_RECORD_LINK``` by the way. Will add a proper review after you had some time to see if I understood your function correctly